src/main/scala/io/qbeast/spark/delta/SamplingListFilesStrategy.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2021 Qbeast Analytics, S.L.
+ */
+package io.qbeast.spark.delta
+
+import io.qbeast.spark.index.query.QueryExecutor
+import io.qbeast.spark.index.query.QuerySpecBuilder
+import io.qbeast.spark.utils.TagUtils
+import org.apache.hadoop.fs.FileStatus
+import org.apache.hadoop.fs.Path
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.expressions.GenericInternalRow
+import org.apache.spark.sql.delta.actions.AddFile
+import org.apache.spark.sql.delta.files.TahoeLogFileIndex
+import org.apache.spark.sql.delta.Snapshot
+import org.apache.spark.sql.execution.datasources.FileStatusWithMetadata
+import org.apache.spark.sql.execution.datasources.PartitionDirectory
+import org.apache.spark.sql.execution.SQLExecution
+
+/**
+ * Implementation of ListFilesStrategy to be used when the query contains sampling clauses.
+ */
+private[delta] object SamplingListFilesStrategy
+    extends ListFilesStrategy
+    with Logging
+    with Serializable {
+
+  override def listFiles(
+      target: TahoeLogFileIndex,
+      partitionFilters: Seq[Expression],
+      dataFilters: Seq[Expression]): Seq[PartitionDirectory] = {
+    val snapshot = target.getSnapshot
+    val path = target.path
+    val files = listStagingAreaFiles(target, partitionFilters, dataFilters) ++
+      listIndexFiles(snapshot, path, partitionFilters, dataFilters)
+    logFilteredFiles(target, snapshot, files)
+    Seq(PartitionDirectory(new GenericInternalRow(Array.empty[Any]), files))
+  }
+
+  private def listStagingAreaFiles(
```

Comment:
Shouldn't this method be common for `ListFilesStrategy`?

Reply:
No, it should not, it only selects the staging area files, it cannot be reused in DefaultListFileStrategy, because there I just use Delta for all the files (it is assumed that there is no replication)